exonum/benches/criterion/storage.rs

```diff
@@ -45,58 +52,119 @@ fn generate_random_kv(len: usize) -> Vec<([u8; KEY_SIZE], Vec<u8>)> {
             rng.fill_bytes(&mut k);
         }
         exists_keys.insert(k);
-        (k, v)
+        (Hash::new(k), v)
     };
 
     (0..len).map(kv_generator).collect::<Vec<_>>()
 }
 
-fn merkle_table_insertion<T: Database>(b: &mut Bencher, db: &T) {
+fn proof_list_append<T: Database>(b: &mut Bencher, db: &T, len: &usize) {
     let mut rng = XorShiftRng::from_seed([200; 16]);
     let mut storage = db.fork();
     let mut table = ProofListIndex::new(NAME, &mut storage);
 
-    b.iter(|| {
-        let v_generator = |_| {
-            let mut chunk = vec![0; 10];
-            rng.fill_bytes(&mut chunk);
-            chunk
-        };
-
-        for item in (0..1000).map(v_generator) {
-            table.push(item);
-        }
-    });
+    b.iter_with_setup(
+        || {
+            (0..*len)
+                .map(|_| {
+                    let mut chunk = vec![0; 64];
+                    rng.fill_bytes(&mut chunk);
+                    chunk
+                }).collect::<Vec<_>>()
+        },
+        |data| {
+            for item in data {
+                table.push(item);
+            }
+        },
+    );
+}
+
+fn proof_map_insert_without_merge<T: Database>(b: &mut Bencher, db: &T, len: &usize) {
+    let data = generate_random_kv(*len);
```

Comment:
Nit: Is there a particular reason why `proof_list_append` generates table contents on the setup stage, but here it's generated once? It looks like both approaches are viable, but I'm for consistency.